benchmark/main.go

```diff
@@ -100,6 +103,9 @@ func main() {
 			return
 		}
 
+		// TODO: Add a more concrete way to check if the rules are updated or not
```

Comment:
I need some suggestion for this @alban @mauriciovasquezbernal 

What would be the appropriate time to wait. Since this doesn't affect the setup time that we measure, it only affects the running time of the test.

Reply:
The `filter` interface could have a `WaitUntilReady()` method.

For calico, you could execute a command `ipset list` in a loop every second until you notice you have the right number of entries:

```
$ sudo ipset list myset-ip -terse|grep '^Number of entries'
Number of entries: 1
```

For bpf/ipsets/iptables, it could be a no-op.

I think that for calico, it would be good to measure the time until the list is really enforced, rather than just measuring the `kubectl apply` time. With the above method, the granularity of the measure would be ~1 second, but that could give an idea.

Reply:
the name of the ipset that is generated is not constant in the case of calico.

For `ipset` filter the set name was hardcorded to `myset`. However in the case of the although the name remains the constant for the entire duration (no of rules: 10,100,1000 ..etc). It is not exposed to the outside.

It is generated in the form of `cali40s:UobsDjk3wjc5vYadHLW9RK5`

Reply:
Right. I guess you can still test without specifying the name, and only one will likely have the right amount of IPs.
```
$ sudo ipset list|grep '^Number of entries'
Number of entries: 10000
Number of entries: 0
```

Reply:
@alban I am able to proceed further with the following operations:

1. Before applying the calico resources, get the `ipset list -n`
2. Again query `ipset list -n`, post applying the calico resources
3. Diff of the output is the name of the ipset created by calico.
4. Query to get the number of entries as per your suggestion.
5. check if the numbers match as expected or try step 4 again.

However, in some cases I am still getting some behavior when even though the number of entries match as we proceed to check if the test ping fails or not (it should fail). I believe this could be the updating of the iptables entry to match the ipset.

Reply:
> I believe this could be the updating of the iptables entry to match the ipset.

It seems likely... I guess you will have to check the result of "iptables ... | grep cali40s..."